packages/runtime/runtime-utils/src/summarizerNode/summarizerNode.ts

```diff
@@ -169,7 +169,7 @@ export class SummarizerNode implements IRootSummarizerNode {
         const summary = new SummaryNode({
             ...localPathsToUse,
             referenceSequenceNumber: this.wipReferenceSequenceNumber,
-            basePath: parentPath,
+            basePath: parentSkipRecursion ? this._latestSummary?.basePath ?? parentPath : parentPath,
```

Comment:
This fix is very specific to this particular scenario. I believe we need a generic fix that fixes the root-cause because it has the potential to break in other places as well.
The root-cause is that the following invariant is broken - A data store cannot be realized during the summarization process after it is summarized.

I was looking at this code and I realized that the `if (parentSkipRecursion)` condition above was added for handling cases where a data store did a partial summary when it failed to summarize. Basically, if a data store failed to summarize, it would send a pointer to the previous summary + all ops after that summary. However, we removed that logic as it was buggy and so, we don't need this logic as well. We can simply say that if `parentSkipRecursion` is true, don't do anything and return.

Reply:
@NicholasCouri Can you create an issue to remove the unncessary code as per my comment here?

Reply:
sure thing.